src/App.tsx

```diff
@@ -15,36 +15,83 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = 'default',
+  Alphabetically = 'alphabetically',
+  Length = 'length',
+}
+
 export const App: React.FC = () => {
+  const [sortMode, setSortMode] = useState<SortType>(SortType.Default);
+  const [isReverse, setIsReverse] = useState<boolean>(false);
+  const sorteredGoods: string[] = [...goodsFromServer].sort((a, b) => {
+    if (sortMode === SortType.Alphabetically) {
+      return a.localeCompare(b);
+    } else if (sortMode === SortType.Length) {
+      return a.length - b.length;
+    }
+
+    return 0;
+  });
+
+  const finalGoods: string[] = isReverse
+    ? [...sorteredGoods].reverse()
+    : sorteredGoods;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${!(sortMode === SortType.Alphabetically) && 'is-light'}`}
```

Comment:
To set classes conditionally use [classnames](https://www.npmjs.com/package/classnames) or [clsx](https://www.npmjs.com/package/clsx) library